bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/handler/VehicleHandler.java

```diff
@@ -230,8 +241,9 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     private State getTripValue(String channelId, TripDetail tripDetails) {
         switch (channelId) {
             case TRIP_CONSUMPTION:
-                if (tripDetails.fuelConsumption != null) {
-                    return new QuantityType<>(tripDetails.fuelConsumption.floatValue() / 100, SmartHomeUnits.LITRE);
+                if (tripDetails.getFuelConsumption().isPresent()) {
+                    return new QuantityType<>(tripDetails.getFuelConsumption().get().floatValue() / 100,
+                            SmartHomeUnits.LITRE);
                 } else {
                     return UnDefType.UNDEF;
                 }
```

Comment:
You should take advantage of having optionals available:
```suggestion
                return tripDetails.getFuelConsumption().map(value -> new QuantityType<>(value.floatValue() / 100,
                            SmartHomeUnits.LITRE).orElse(UnDefType.UNDEF);
```

Reply:
This does not work. My usage of Optional is not extensive as it could, I must get used to it.

Reply:
Looks like my suggestion was missing a parenthesis.

Reply:
Still fails : The method orElse(QuantityType<Volume>) in the type Optional<QuantityType<Volume>> is not applicable for the arguments (UnDefType)

Reply:
But this seems ok :                 return tripDetails.getFuelConsumption()
                        .map(value -> (State) new QuantityType<>(value.floatValue() / 100, SmartHomeUnits.LITRE))
                        .orElse(UnDefType.UNDEF);